script.js

```diff
@@ -67,36 +57,33 @@ class IntBuilder extends Builder {
 //   .divide(4)                         // 7;
 //   .mod(3)                            // 1;
 //   .get();                            // -> 1;
-// console.log(intBuilder.x);
+// console.log(intBuilder);
 // console.log(IntBuilder.random(10, 100));
 
 
 /////// ES5 for strings
 function StringBuilder(x) {
   Builder.call(this, x);
-  this.x = x||'';
+  this.x = x || '';
 }
 
 StringBuilder.prototype = Object.create(Builder.prototype);
 
 StringBuilder.prototype.sub = function(from, n) {
-  this.from = from;
-  this.n = n;
-  this.x = this.x.substr(this.from, this.n);
+  this.x = this.x.substr(from, n);
   return this;
 }
 
 StringBuilder.prototype.remove = function(str) {
-  this.str = str;
-  let from = this.x.indexOf(this.str);
+  let from = this.x.indexOf(str);
   let to = from + str.length;
   if (from == -1) {
     return;
   } else {
-    this.x = `${this.x.slice(0, from)}${this.x.slice(to)}`;
+    this.x = this.x.slice(0, from) + this.x.slice(to);
     return this;
   }
-}
+}//it is said in the task not to use replace(), is there one more simple solution?
```

Comment:
Oh, sorry. You are right here. It was my mistake.

Reply:
It`s okey